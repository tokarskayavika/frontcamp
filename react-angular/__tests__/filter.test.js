import React from 'react';
import Enzyme from 'enzyme';
import Filter from '../scripts/ui/posts/Filter';
import Adapter from 'enzyme-adapter-react-16';
import expectFrom from 'expect';
import enzymify from 'expect-enzyme';
import renderer from 'react-test-renderer';

Enzyme.configure({
    adapter: new Adapter()
});

expectFrom.extend(enzymify());

describe('Filter Component', () => {
    let filter;

    beforeEach(() => {
        filter = Enzyme.shallow(<Filter changeFilterValue={() => {}} />);
    });

    it('component should be a div', () => {
        expectFrom(filter).toBeA('div');
    });

    it('filter should contain an input', () => {
        expectFrom(filter).toContain('input');
    });

    it('filter should not contain a section element', () => {
        expectFrom(filter).toNotContain('section');
    });

    it('should be a label inside', () => {
        expectFrom(filter.find('label')).toHaveRendered();
    });

    it('filter renders properly', () => {
        const filter = renderer
            .create(<Filter changeFilterValue={() => {}} />)
            .toJSON();
        expect(filter).toMatchSnapshot();
    });
});