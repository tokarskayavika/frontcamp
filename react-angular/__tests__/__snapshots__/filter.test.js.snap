// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component filter renders properly 1`] = `
<div
  className="filter"
>
  <input
    id="filter-input"
    onChange={[Function]}
    type="text"
  />
  <label
    htmlFor="filter-input"
  >
    Filter by author
  </label>
</div>
`;
